src/CommunityToolkit.Maui/Views/Popup/Popup.shared.cs

```diff
@@ -352,6 +352,10 @@ internal override void OnParentResourcesChanged(IEnumerable<KeyValuePair<string,
 	protected virtual async Task OnClosed(object? result, bool wasDismissedByTappingOutsideOfPopup)
 	{
 		((IPopup)this).OnClosed(result);
+		if (resources is not null)
+		{
+			((IResourceDictionary)resources).ValuesChanged -= OnResourcesChanged;
```

Comment:
Do we need to cast here? If so it might be clearer to modify the if statement to pattern match e.g.

```csharp
if (resources is IResourceDictionary resourceDictionary)
{
    resourceDictionary.ValuesChanged -= OnResourcesChanged;
}
```

What do you think?

Reply:
Oh I see the interface is internal and we are casting in the constructor so we are consistent. Maybe we ignore what I just said as this will catch issues if the API changes